cmd/kubeadm/app/util/token/tokens.go

```diff
@@ -43,13 +42,20 @@ var (
 	TokenRegexp = regexp.MustCompile(TokenRegexpString)
 )
 
+const validBootstrapTokenChars = "0123456789abcdefghijklmnopqrstuvwxyz"
+
 func randBytes(length int) (string, error) {
 	b := make([]byte, length)
 	_, err := rand.Read(b)
 	if err != nil {
 		return "", err
 	}
-	return hex.EncodeToString(b), nil
+
+	for i, n := range b {
+		b[i] = validBootstrapTokenChars[int(n)%len(validBootstrapTokenChars)]
```

Comment:
This is much better than before, but will still less than the full amount of entropy.

It has an uneven distribution of output characters because `256 mod len("0123456789abcdefghijklmnopqrstuvwxyz") = 256 mod 36 = 4`, so uniformly distributed input bytes will get placed into non-uniform output buckets. The first few characters `0123` will occur more than you'd expect.

Reply:
There is probably a faster way to do this with `rand.Read()` and rejection sampling, but this seems to work:
```go
// import "math/big"
// import "crypto/rand"

func randBytes(length int) (string, error) {
	result := make([]byte, length)
	maxIdx := big.NewInt(int64(len(validBootstrapTokenChars)))
	for i := range result {
		idx, err := rand.Int(rand.Reader, maxIdx)
		if err != nil {
			return "", err
		}
		result[i] = validBootstrapTokenChars[idx.Int64()]
	}
	return string(result), nil
}
```

Reply:
Ah, you're right. Updated to discard bytes >= 252.

Reply:
Woops. The GitHub UI didn't show your second comment to me. Did something similar but avoided using "math/big"